### PR TITLE
[FIX] Livechat user registration in another department

### LINKED ISSUE #7398: LiveChat User registration
Rocket.Chat Version: 0.56.0
Running Instances: 1
DB Replicaset OpLog: Enabled
Node Version: v4.8.1

If user is registered in a particular department and conversation is closed.
Second time user with same username and email cant login into another department.
Agent from previous department is assigned again, it's like user login to previous department without considering department parameter.

### PR DESCRIPTION
**Pull request by renatobecker**
Closes #7398 

This PR adds a verification when a registered visitor attempted to register with a different department that they previously registered. If a registered visitor tried to register in another department, it was being sent to the same department / agent and is not a correct behavior.

### CODE REVIEW

File: packages/rocketchat-livechat/.app/imports/client/visitor.js [hunk 1]
```diff
@@ -38,6 +38,11 @@ export default {
 		return this.token.get();
 	},
 
+	getDepartment() {
+		const data = this.getData();
+		return data ? data.department : null;
```
Comment (ggazzo) [inline packages/rocketchat-livechat/.app/imports/client/visitor.js]: we can use ```

Comment (renatobecker) [reply packages/rocketchat-livechat/.app/imports/client/visitor.js]: @ggazzo !
Ternary condition removed.
Thx.

**Comment by sampaiodiego** [comment]
@renatobecker I guess this is still valid right? if so, can you please fix the conflicts? thanks

**Comment by renatobecker** [comment]
@sampaiodiego, done!
Thanks.

File: packages/rocketchat-livechat/.app/client/lib/chatMessages.js [hunk 2]
```diff
@@ -142,7 +142,7 @@ this.ChatMessages = class ChatMessages {
 			});
 		};
 
-		if (!visitor.getId()) {
+		if ((!visitor.getId()) || (visitor.getDepartment() !== Livechat.department)) {
```
Comment (sampaiodiego) [inline packages/rocketchat-livechat/.app/client/lib/chatMessages.js]: please avoid unnecessary parenthesis
```suggestion
		if (!visitor.getId() || visitor.getDepartment() !== Livechat.department) {
```

**Pull request merged**